### PR TITLE
ENG-62: implement collectionAttemptMachine definition + state × event matrix test

### PR DESCRIPTION
**Pull request by Connorbelez**
ENG-62: implement collectionAttemptMachine definition + state × event matrix test

Replace placeholder machine (v0.1.0) with full 7-state, 8-event collection
attempt machine (v1.0.0) per SPEC 1.5 §4.1. Includes canRetry guard,
6 effect-marker action stubs, and typed context (retryCount/maxRetries).

Add exhaustive test suite: 56 state × event matrix cases, 3 guard boundary
tests, 6 happy path integration tests (71 total). Covers ManualPaymentMethod
immediate path, async PAD path, retry with backoff, and terminal lockdown.

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

docs

### CODE REVIEW

**Comment by Connorbelez** [comment]
* **#121** <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/121?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Connorbelez/tanstackTemplate/121?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by Connorbelez** [comment]
@coderabbitai please review

File: convex/engine/machines/collectionAttempt.machine.ts [hunk 1]
```diff
@@ -1,24 +1,107 @@
...
+			/* resolved by GT effect registry */
+		},
+		emitPaymentReceived: () => {
+			/* resolved by GT effect registry */
+		},
+		incrementRetryCount: () => {
+			/* resolved by GT effect registry */
+		},
+		scheduleRetryEntry: () => {
+			/* resolved by GT effect registry */
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/engine/machines/collectionAttempt.machine.ts]: Fixed: Changed `incrementRetryCount` from a no-op effect-marker stub to an XState `assign` action that increments `context.retryCount`. This ensures the `canRetry` guard evaluates against the correct value during pure state computation — `assign` runs in the COMPUTE step and the updated context is persisted atomically by the transition engine via `nextSnapshot.context`. Other actions remain as effect-marker stubs since they trigger side effects (not context mutations).

File: convex/engine/machines/__tests__/collectionAttempt.test.ts [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+		});
+
+		it("pending -> failed on DRAW_FAILED fires incrementRetryCount", () => {
+			const [next, actions] = transition(
+				collectionAttemptMachine,
+				snapshotAt("pending"),
+				DRAW_FAILED
+			);
+			expect(next.value).toBe("failed");
+			expect(actions.map((a) => a.type)).toContain("incrementRetryCount");
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/engine/machines/__tests__/collectionAttempt.test.ts]: Fixed: Updated the `pending → failed` test and the async failure happy path test to assert `next.context.retryCount === 1` after DRAW_FAILED. Since `incrementRetryCount` now uses XState's built-in `assign`, the context mutation happens during computation and is reflected in `nextSnapshot.context` — so we verify the result rather than checking for an action name in the actions array.
